src/reducers/archiveThought.js

```diff
@@ -67,7 +68,8 @@ export default (state, { path } = {}) => {
   const isEmpty = value === ''
   const isArchive = value === '=archive'
   const isArchived = isThoughtArchived(path)
-  const isDeletable = isEmpty || isArchive || isArchived
+  const hasDescendants = getDescendants(state, path).length !== 0
+  const isDeletable = (isEmpty || isArchive || isArchived) && !hasDescendants
```

Comment:
I would use `getThoughts` here. It is much faster than `getDescendants`, and has the same effect in this case.